packages/peregrine/src/hooks/__tests__/useDocumentListener.spec.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { act } from 'react-test-renderer';
+
+import { useDocumentListener } from '../useDocumentListener';
+import createTestInstance from '../../util/createTestInstance';
+
+const spies = new Map();
+const handleClick = jest.fn();
+
+spies.set(
+    'addEventListener',
+    jest.spyOn(document, 'addEventListener').mockImplementation(jest.fn())
+);
+spies.set(
+    'removeEventListener',
+    jest.spyOn(document, 'removeEventListener').mockImplementation(jest.fn())
+);
+
+const Component = () => {
+    useDocumentListener('click', handleClick, { once: true });
+
+    return <i />;
+};
+
+test('adds an event listener to the document on mount', () => {
+    createTestInstance(<Component />);
+
+    const spy = spies.get('addEventListener');
```

Comment:
Oh, I just did this to avoid naming struggles since there's more than one spy.

```es6
// 🙂 
const spy = jest.spyOn(...)

// 🤔 
const addEventListenerSpy = jest.spyOn(...)
const removeEventListenerSpy = jest.spyOn(...)

// 😌 
const spies = new Map()
  .set("addEventListener", jest.spyOn(...))
  .set("removeEventListener", jest.spyOn(...))
```

Reply:
For future reference: you don't even need to capture the return value of `jest.spyOn()`, because it returns the same spy object that it sets. So you can just run

```js
jest.spyOn(document.addEventListener)

test(() => {
  expect(document.addEventListener).toHaveBeenCalledTimes(1);
});
```